File: programmers/12905.js
function solution(board) {
  const row = board.length;
  const column = board[0].length;

  let maxLength = board[0][0];

  for (let i = 1; i < row; i++) {
    for (let j = 1; j < column; j++) {
      if (board[i][j] >= 1) {
        const cross = board[i - 1][j - 1];
        const up = board[i - 1][j];
        const left = board[i][j - 1];

        if (cross >= 1 && up >= 1 && left >= 1) {
          board[i][j] = Math.min(cross, up, left) + 1;
        }
        maxLength = Math.max(maxLength, board[i][j]);
      }
    }
  }
  return maxLength ** 2;
}

File: programmers/12905.py
from typing import List


def solution(board: List[List[int]]) -> int:
    answer = 1 if 1 in board[0] else 0
    sides = []

    for i in range(len(board)):
        for j in range(len(board[0])):
            if i == 0 or j == 0:
                continue

            if board[i][j] > 0:
                board[i][j] = min(board[i - 1][j], board[i][j - 1], board[i - 1][j - 1]) + 1
        sides.append(max(board[i]))

    if len(sides) <= 0:
        return answer

    max_side = max(sides)
    return max_side**2
